### PR TITLE
Implement invite user to enterprise

### PR DESCRIPTION
**Pull request by mattwiller**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @mattwiller has signed the CLA. Thanks for the pull request!

**Comment by mattwiller** [comment]
Fixes #61

File: src/test/java/BoxInviteTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+/**
+ * {@link BoxInvite} related tests.
+ */
+public class BoxInviteTest {
```
Comment (iamharish) [inline src/test/java/BoxInviteTest.java]: Would you be able to add an integration test as this is a new api?

**Comment by mattwiller** [comment]
@iamharish This would be kind of tricky to add integration tests for, since it involves setting up a user account that isn't in the enterprise.  It's not easy to create that through the API today — certainly not in a way that would be easily repeatable.  Do you have any ideas on how we could set up an end-to-end test for this?

**Pull request merged**